envoy-control-tests/src/main/resources/lua_spec/ingress_rbac_logging_spec.lua

```diff
@@ -60,6 +60,25 @@ describe("envoy_on_request:", function()
         assert.spy(metadata.set).was_called_with(_, "envoy.filters.http.lua", "request.info.xff_header", "127.0.4.3")
     end)
 
+    it("should set dynamic metadata for request id", function()
+        -- given
+        local headers = {
+            ['x-custom-request-id'] = '123-456-789',
+        }
+        local filter_metadata = {
+            ['request_id_headers'] = { 'x-custom-request-id' }
+        }
+
+        local handle = handlerMock(headers, {}, nil, filter_metadata)
+        local metadata = handle:streamInfo():dynamicMetadata()
+
+        -- when
+        envoy_on_request(handle)
+
+        -- then
+        assert.spy(metadata.set).was_called_with(_, "envoy.filters.http.lua", "request.info.request_id", "123-456-789")
+    end)
+
```

Comment:
I think we are missing test with actual log.

Reply:
If you mean an integration test then I added it.